src/test/java/project/mybookshop/service/BookServiceImplTest.java

```diff
@@ -0,0 +1,62 @@
+package project.mybookshop.service;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.math.BigDecimal;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.annotation.Commit;
+import org.springframework.test.annotation.Rollback;
+import org.springframework.transaction.annotation.Transactional;
+import project.mybookshop.dto.BookDto;
+import project.mybookshop.dto.CreateBookRequestDto;
+import project.mybookshop.exceptions.DataProcessingException;
+import project.mybookshop.mapper.BookMapper;
+
+@Transactional
+@SpringBootTest
+public class BookServiceImplTest {
+    @Autowired
+    private BookService bookService;
+    @Autowired
+    private BookMapper bookMapper;
+    private CreateBookRequestDto testBookRequestDto;
+
+    @BeforeEach
+    void setUp() {
+        testBookRequestDto = new CreateBookRequestDto();
+        testBookRequestDto.setTitle("Test book");
+        testBookRequestDto.setAuthor("Test author");
+        testBookRequestDto.setIsbn("1111");
+        testBookRequestDto.setPrice(BigDecimal.valueOf(22));
+        testBookRequestDto.setDescription("Test book!");
+        testBookRequestDto.setCoverImage("test.jpg");
+    }
+
+    @Rollback
+    @Test
+    void save_checkOnRollback() {
+        try {
+            bookService.save(testBookRequestDto);
+        } catch (DataProcessingException e) {
+            assertTrue(e.getMessage().contains("Can't insert book into DB: "));
```

Comment:
for now it's ok that you check exception by their's message, but in larger projects it would be a bad practice, because you can change exception message in place where you throw it, and tests would fail